### ISSUE TITLE
Android crash with the latest version

### ISSUE THREAD

**Issue by viniychuk** [blackuy/react-native-twilio-video-webrtc#51]
Hi, Thanks so much for your work — I'm using the package on iOS and it works flawlessly!
Finally got to Android setup and it's crashing when I'm trying to get to the room.
Appreciate your help and ready to assist with any debug information!

### Steps to reproduce
1. `react-native init twilioTest`
2. `yarn add https://github.com/blackuy/react-native-twilio-video-webrtc`
3. Followed the steps from the manual to link the library
4. Run the Android test app (basically I took the content of `index.ios.js` and put it into `index.android.js` since there's no platform related staff.
5. Put the token and room name into the fields and press "Connect"

### Expected behaviour
View with my video (room was empty)

### Actual behaviour
Crashes with an error — http://take.ms/i7cB9

### Environment
- **Node.js version**: 8.4.0
- **React Native**: 0.44.0
- **React Native Cli**: 2.0.1
- **React Native platform + platform version**: Android 5.0

### react-native-twilio-video-webrtc
**Version**: react-native-twilio-video-webrtc@1.0.2

**Comment by viniychuk** [comment]
Update: I found an issue and fixed it — it's a RECORD_AUDIO permission thing. I did it in the `MainActivity` so now I'm thinking on how to make a PR to let other people now how it works...

**Comment by bilby91** [comment]
Great that you solved your problem @viniychuk ! You did something different from what is instructed in the README ?

**Comment by viniychuk** [comment]
@bilby91 Yes, basically the problem is that in the newer versions of Android you have to manually ask for permissions... although when you generate the signed APK it works from the very beginning. I ended writing a bit of Native Java code but later discovered that there's a react-native package that allow you to do just that. I'll work with it on the weekend and submit a PR to update README instruction.

**Comment by bilby91** [comment]
@viniychuk Awesome! Having some information on the README for newer versions of Android would be great. What version started requiring the permission ?

**Comment by viniychuk** [comment]
@bilby91 Android 6.0

**Comment by slycoder** [comment]
Did you have to do anything beyond what is included in the readme under `android` ... `permissions`?  Hopefully no native code is required beyond this package as `PermissionsAndroid` should take care of it.

We should probably unify the examples at some point though.

**Comment by akshay-squareboat** [comment]
> @bilby91 Yes, basically the problem is that in the newer versions of Android you have to manually ask for permissions... although when you generate the signed APK it works from the very beginning. I ended writing a bit of Native Java code but later discovered that there's a react-native package that allow you to do just that. I'll work with it on the weekend and submit a PR to update README instruction.

Can You Share the code you have written to grant the audio and video permissions for this App?

**Comment by slycoder** [comment]
There's a description here: https://github.com/blackuy/react-native-twilio-video-webrtc#permissions-1 of what you need to do if you haven't set up permissions yet.  Here is a link to the RN functionality for managing permissions: http://reactnative.dev/docs/permissionsandroid.html
